lib/std/os/uefi/protocols/device_path_protocol.zig

```diff
@@ -204,37 +280,71 @@ pub const AcpiDevicePath = union(Subtype) {
         _,
     };
 
-    pub const BaseAcpiDevicePath = packed struct {
+    pub const BaseAcpiDevicePath = extern struct {
         type: DevicePathType,
         subtype: Subtype,
-        length: u16,
-        hid: u32,
-        uid: u32,
+        length: u16 align(1),
+        hid: u32 align(1),
+        uid: u32 align(1),
     };
 
-    pub const ExpandedAcpiDevicePath = packed struct {
+    comptime {
+        std.debug.assert(12 == @sizeOf(BaseAcpiDevicePath));
+        std.debug.assert(1 == @alignOf(BaseAcpiDevicePath));
+
+        std.debug.assert(0 == @offsetOf(BaseAcpiDevicePath, "type"));
+        std.debug.assert(1 == @offsetOf(BaseAcpiDevicePath, "subtype"));
+        std.debug.assert(2 == @offsetOf(BaseAcpiDevicePath, "length"));
+        std.debug.assert(4 == @offsetOf(BaseAcpiDevicePath, "hid"));
+        std.debug.assert(8 == @offsetOf(BaseAcpiDevicePath, "uid"));
+    }
+
+    pub const ExpandedAcpiDevicePath = extern struct {
         type: DevicePathType,
         subtype: Subtype,
-        length: u16,
-        hid: u32,
-        uid: u32,
-        cid: u32,
+        length: u16 align(1),
+        hid: u32 align(1),
+        uid: u32 align(1),
+        cid: u32 align(1),
         // variable length u16[*:0] strings
         // hid_str, uid_str, cid_str
     };
 
-    pub const AdrDevicePath = packed struct {
+    comptime {
+        std.debug.assert(16 == @sizeOf(ExpandedAcpiDevicePath));
+        std.debug.assert(1 == @alignOf(ExpandedAcpiDevicePath));
+
+        std.debug.assert(0 == @offsetOf(ExpandedAcpiDevicePath, "type"));
+        std.debug.assert(1 == @offsetOf(ExpandedAcpiDevicePath, "subtype"));
+        std.debug.assert(2 == @offsetOf(ExpandedAcpiDevicePath, "length"));
+        std.debug.assert(4 == @offsetOf(ExpandedAcpiDevicePath, "hid"));
+        std.debug.assert(8 == @offsetOf(ExpandedAcpiDevicePath, "uid"));
+        std.debug.assert(12 == @offsetOf(ExpandedAcpiDevicePath, "cid"));
+    }
+
+    pub const AdrDevicePath = extern struct {
         type: DevicePathType,
         subtype: Subtype,
-        length: u16,
-        adr: u32,
+        length: u16 align(1),
+        adr: u32 align(1),
+
         // multiple adr entries can optionally follow
         pub fn adrs(self: *const AdrDevicePath) []const u32 {
             // self.length is a minimum of 8 with one adr which is size 4.
             var entries = (self.length - 4) / @sizeOf(u32);
-            return @ptrCast([*]const u32, &self.adr)[0..entries];
+            return @ptrCast([*]const u32, @alignCast(4, &self.adr))[0..entries];
```

Comment:
The struct declaration is align(1) because all DevicePaths are laid out in memory right after eachother with no padding. I'm actually leaning towards adrs being align(1) as well because the spec only says they are 4 bytes, not that they are u32

Reply:
If the struct is meant to be align(1), then the alignCast is wrong.

Reply:
Yeah, I just changed this to return `[]align(1) const u32`, given I'm pretty sure that UEFI doesn't give us any alignment guarantees here